dbms/src/Flash/tests/gtest_compute_server.cpp

```diff
@@ -445,5 +501,115 @@ try
     }
 }
 CATCH
+
+/// For FineGrainedShuffleJoin/Agg test usage, update internal exchange senders/receivers flag
+/// Allow select,agg,join,tableScan,exchangeSender,exchangeReceiver executors only
+void setFineGrainedShuffleForExchange(tipb::Executor & root)
+{
+    tipb::Executor * current = &root;
+    while (current)
+    {
+        switch (current->tp())
+        {
+        case tipb::ExecType::TypeSelection:
+            current = const_cast<tipb::Executor *>(&current->selection().child());
+            break;
+        case tipb::ExecType::TypeAggregation:
+            current = const_cast<tipb::Executor *>(&current->aggregation().child());
+            break;
+        case tipb::ExecType::TypeJoin:
+        {
+            /// update build side path
+            JoinInterpreterHelper::TiFlashJoin tiflash_join{current->join()};
+            current = const_cast<tipb::Executor *>(&current->join().children()[tiflash_join.build_side_index]);
+            break;
+        }
+        case tipb::ExecType::TypeExchangeSender:
+            if (current->exchange_sender().tp() == tipb::Hash)
+                current->set_fine_grained_shuffle_stream_count(8);
```

Comment:
We can modify the `AggregationBinder::toMPPSubPlan` to pass the `fine_grained_shuffle_stream_count` to `ExchangeReceiverBinder` and `ExchangeSenderBinder`.
Then the `ExchangexxxBinder` can use `fine_grained_shuffle_stream_count` to set the `tipb::Executor`

Reply:
> We can modify the `AggregationBinder::toMPPSubPlan` to pass the `fine_grained_shuffle_stream_count` to `ExchangeReceiverBinder` and `ExchangeSenderBinder`. Then the `ExchangexxxBinder` can use `fine_grained_shuffle_stream_count` to set the `tipb::Executor`

I will refactor it when this pr merged.